starbucks_dw/models/reporting/customer_offer_responses.sql

```diff
@@ -0,0 +1,45 @@
+WITH offer_details AS (
+  SELECT
+    offer_id,
+    offer_type,
+    channel as offer_channel,
+    difficulty_rank,
+    duration,
+    reward
+  FROM {{ ref('dim_offer') }}
+),
+
+customer_transactions AS (
+  SELECT
+    transaction_id,
+    customer_id,
+    offer_id,
+    transaction_status
+  FROM {{ ref('fct_customer_transactions') }}
+)
+
+SELECT
+  ct.customer_id,
+  ct.transaction_id,
+  od.offer_id,
+  od.offer_type,
+  od.offer_channel,
+  od.difficulty_rank,
+  od.duration,
+  od.reward,
+  ct.transaction_status,
+  ft.days_since_start,
+  ft.hours_since_start,
+  CASE WHEN ct.transaction_status = 'Responded' THEN 1 ELSE 0 END as response_flag,
```

Comment:
Not sure where you did find `Responded` transaction status 🤔  The values for transaction_status column are: `completed`, `viewed` and `received`. Are you doing any transformation?